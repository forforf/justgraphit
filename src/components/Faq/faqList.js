// Strange form due to porting from Angular
const faqList = [
  [
    'Where is my data stored?',
    'The data is stored on the local device, the network never sees it.',
  ],
  [
    'Do I need a network connection?',
    'Only to get to the initial page, after that no network is required.',
  ],
  ['How much is it?', 'Free.'],
  ["I don't see any ads?", 'No Ads.'],
  [
    'How was this built?',
    'Originally on Angular1, then Angular2, then classic React, now functional React',
  ],
  [
    'How do I use it?',
    "Enter a number you want plotted on the graph page. That's it. To create a new graph, just change the name.",
  ],
];
export default faqList;
