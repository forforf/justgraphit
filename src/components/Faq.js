import React from 'react';
import {List, ListItem, ListItemText} from '@mui/material';
import faqList from './Faq/faqList';

const listItems = faqList.map((qa, i) => (
  <ListItem key={i}>
      <ListItemText primary={qa[0]} secondary={qa[1]} />
  </ListItem>
));
const Faq = () => (
    <List>
      { listItems }
    </List>
);

export default Faq;
