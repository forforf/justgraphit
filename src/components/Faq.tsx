import React from 'react';
import { List, ListItem, ListItemText } from '@mui/material';
import faqList from './Faq/faqList';

const listItems = faqList.map((qa: string[], i) => (
  <ListItem key={i}>
    <ListItemText primary={qa[0]} secondary={qa[1]} />
  </ListItem>
));

const Faq = (): JSX.Element => <List>{listItems}</List>;

export default Faq;
