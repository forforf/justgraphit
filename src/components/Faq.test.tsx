import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Faq from './Faq';

test('deep snapshot', () => {
  const component = renderer.create(<Faq />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('shallow snapshot', () => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(<Faq />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
