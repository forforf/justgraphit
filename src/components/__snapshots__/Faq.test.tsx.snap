// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep snapshot 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-1.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-1.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-1.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-1.Mui-disabled {
  opacity: 0.38;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 6px;
  margin-bottom: 6px;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: block;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: rgba(0, 0, 0, 0.6);
  display: block;
}

<ul
  className="MuiList-root MuiList-padding emotion-0"
>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline emotion-2"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
      >
        Where is my data stored?
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-4"
      >
        The data is stored on the local device, the network never sees it.
      </p>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline emotion-2"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
      >
        Do I need a network connection?
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-4"
      >
        Only to get to the initial page, after that no network is required.
      </p>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline emotion-2"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
      >
        How much is it?
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-4"
      >
        Free.
      </p>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline emotion-2"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
      >
        I don't see any ads?
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-4"
      >
        No Ads.
      </p>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline emotion-2"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
      >
        How was this built?
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-4"
      >
        Originally on Angular1, then Angular2, then classic React, now functional React
      </p>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline emotion-2"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
      >
        How do I use it?
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-4"
      >
        Enter a number you want plotted on the graph page. That's it. To create a new graph, just change the name.
      </p>
    </div>
  </li>
</ul>
`;

exports[`shallow snapshot 1`] = `
<ForwardRef(List)>
  <ForwardRef(ListItem)>
    <ForwardRef(ListItemText)
      primary="Where is my data stored?"
      secondary="The data is stored on the local device, the network never sees it."
    />
  </ForwardRef(ListItem)>
  <ForwardRef(ListItem)>
    <ForwardRef(ListItemText)
      primary="Do I need a network connection?"
      secondary="Only to get to the initial page, after that no network is required."
    />
  </ForwardRef(ListItem)>
  <ForwardRef(ListItem)>
    <ForwardRef(ListItemText)
      primary="How much is it?"
      secondary="Free."
    />
  </ForwardRef(ListItem)>
  <ForwardRef(ListItem)>
    <ForwardRef(ListItemText)
      primary="I don't see any ads?"
      secondary="No Ads."
    />
  </ForwardRef(ListItem)>
  <ForwardRef(ListItem)>
    <ForwardRef(ListItemText)
      primary="How was this built?"
      secondary="Originally on Angular1, then Angular2, then classic React, now functional React"
    />
  </ForwardRef(ListItem)>
  <ForwardRef(ListItem)>
    <ForwardRef(ListItemText)
      primary="How do I use it?"
      secondary="Enter a number you want plotted on the graph page. That's it. To create a new graph, just change the name."
    />
  </ForwardRef(ListItem)>
</ForwardRef(List)>
`;
